Supplement README_MACOSX_NOTES

JoeCohen opened the pull request:
- Lists more MySQL vestiges to delete before upgrading to v9.
- Move instructions about `config/db.yml` to place where problem likely first occurs
- Add content of `db.yml`
- More some instruction details to footnotes. 
- Other formatting changes

JoeCohen committed:
Formatting changes, more mysql cleanup

nimmolo commented on README_MACOSX_NOTES.md:
```diff
@@ -31,28 +33,12 @@ brew outdated
 
-Install a bunch of useful stuff from `Homebrew`
+**If you have mysql version < 9.0 installed, remove mysql and all its vestiges before
+continuing. See footnote 1.<sup id="a1">[1](#f1)</sup>**
```
Joe - I'd add a note that if you have older hardware, you should neither update Homebrew nor MySQL, which will not install in any case on older Macs. Following this step was a 12-hour time-waster for me in Virginia.

JoeCohen replied:
Thanks for the suggestion. I added a warning near the top of the file.
I also added a Table of Contents to make the file easier to navigate.
I will merge it.

JoeCohen committed:
Warning re older macs; TOC

- Add warning suggested by @nimmolo re older macs
- Reformat for, and generate, Table of Contents
